streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableTransformValues.java

```diff
@@ -158,7 +158,11 @@ public void init(final ProcessorContext context) {
             internalProcessorContext.setRecordContext(new ProcessorRecordContext(
                 valueAndTimestamp == null ? UNKNOWN : valueAndTimestamp.timestamp(),
                 -1L, // we don't know the original offset
-                currentContext.partition(),
+                // technically, we know the partition, but in the new `api.Processor` class,
+                // we move to `RecordMetadata` than would be `null` for this case and thus
+                // we won't have the partition information, so it's better to now provide it
+                // here either, to not introduce a regression later on
+                -1,
```

Comment:
@vvcephei I think it's better to change this...

Reply:
SGTM, thanks @mjsax to bring to our attention.